src/platform/audio/ffmpegaudiofilecache.cpp

```diff
@@ -0,0 +1,537 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief Class for caching loaded audio samples to reduce file IO.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+extern "C" {
```

Comment:
Why is this `extern "C"` ?

Reply:
Because the headers can't be included from C++. See https://trac.ffmpeg.org/wiki/Including%20FFmpeg%20headers%20in%20a%20C%2B%2B%20application